plugins/database/hana/hana.go

```diff
@@ -84,59 +92,55 @@ func (h *HANA) getConnection(ctx context.Context) (*sql.DB, error) {
 
 // CreateUser generates the username/password on the underlying HANA secret backend
 // as instructed by the CreationStatement provided.
-func (h *HANA) CreateUser(ctx context.Context, statements dbplugin.Statements, usernameConfig dbplugin.UsernameConfig, expiration time.Time) (username string, password string, err error) {
+func (h *HANA) NewUser(ctx context.Context, req newdbplugin.NewUserRequest) (response newdbplugin.NewUserResponse, err error) {
 	// Grab the lock
 	h.Lock()
 	defer h.Unlock()
 
-	statements = dbutil.StatementCompatibilityHelper(statements)
-
 	// Get the connection
 	db, err := h.getConnection(ctx)
 	if err != nil {
-		return "", "", err
+		return newdbplugin.NewUserResponse{}, err
 	}
 
-	if len(statements.Creation) == 0 {
-		return "", "", dbutil.ErrEmptyCreationStatement
+	if len(req.Statements.Commands) == 0 {
+		return newdbplugin.NewUserResponse{}, dbutil.ErrEmptyCreationStatement
 	}
 
+	dispName := credsutil.DisplayName(req.UsernameConfig.DisplayName, 32)
+	roleName := credsutil.RoleName(req.UsernameConfig.RoleName, 20)
+	maxLen := credsutil.MaxLength(maxIdentifierLength)
+	separator := credsutil.Separator("_")
+	caps := credsutil.ToUpper()
+
 	// Generate username
-	username, err = h.GenerateUsername(usernameConfig)
+	username, err := credsutil.GenerateUsername(dispName, roleName, maxLen, separator, caps)
 	if err != nil {
-		return "", "", err
+		return newdbplugin.NewUserResponse{}, err
 	}
 
 	// HANA does not allow hyphens in usernames, and highly prefers capital letters
 	username = strings.Replace(username, "-", "_", -1)
 	username = strings.ToUpper(username)
 
-	// Generate password
-	password, err = h.GeneratePassword()
-	if err != nil {
-		return "", "", err
-	}
 	// Most HANA configurations have password constraints
 	// Prefix with A1a to satisfy these constraints. User will be forced to change upon login
+	password := req.Password
 	password = strings.Replace(password, "-", "_", -1)
```

Comment:
This needs to be removed. Manipulating the password in the plugin is not allowed because Vault won't know about those changes and will report the wrong password to the user.

I notice in the update password function, the default statement includes the password in double quotes. If the creation statement includes double quotes around the password, does this conversion become unnecessary?